diff --git a/lib/Makefile.m32 b/lib/Makefile.m32
index 325cdc7af..773187ad7 100644
--- a/lib/Makefile.m32
+++ b/lib/Makefile.m32
@@ -289,9 +289,14 @@ ifdef ZLIB
   DLL_LIBS += -L"$(ZLIB_PATH)" -lz
 endif
 ifdef BROTLI
-  INCLUDES += -I"$(BROTLI_PATH)"
+  INCLUDES += -I"$(BROTLI_PATH)/include"
   CFLAGS += -DHAVE_BROTLI
-  DLL_LIBS += -L"$(BROTLI_PATH)/lib" -lbrotlidec
+  DLL_LIBS += -L"$(BROTLI_PATH)/lib"
+  ifdef BROTLI_LIBS
+    DLL_LIBS += $(BROTLI_LIBS)
+  else
+    DLL_LIBS += -lbrotlidec
+  endif
 endif
 ifdef IDN2
   INCLUDES += -I"$(LIBIDN2_PATH)/include"
diff --git a/src/Makefile.m32 b/src/Makefile.m32
index ffc359149..91d1b1d95 100644
--- a/src/Makefile.m32
+++ b/src/Makefile.m32
@@ -303,9 +303,14 @@ ifdef ZLIB
   curl_LDADD += -L"$(ZLIB_PATH)" -lz
 endif
 ifdef BROTLI
-  INCLUDES += -I"$(BROTLI_PATH)"
+  INCLUDES += -I"$(BROTLI_PATH)/include"
   CFLAGS += -DHAVE_BROTLI
   curl_LDADD += -L"$(BROTLI_PATH)/lib" -lbrotlidec
+  ifdef BROTLI_LIBS
+    curl_LDADD += $(BROTLI_LIBS)
+  else
+    curl_LDADD += -lbrotlidec
+  endif
 endif
 ifdef IDN2
   CFLAGS += -DUSE_LIBIDN2
